A new example that shows how to integrate Python plugins that use pyclasses into a Rust app (#2873)
PyO3/pyo3: examples/README.md

| `maturin-starter` | A template project which is configured to use [`maturin`](https://github.com/PyO3/maturin) for development. |
| `setuptools-rust-starter` | A template project which is configured to use [`setuptools_rust`](https://github.com/PyO3/setuptools-rust/) for development. |
| `word-count` | A quick performance comparison between word counter implementations written in each of Rust and Python. |

> | `plugin` | Illustrates how to make Python plugins work within a Rust application. Allows to test plugin API separately from the main application. |

Reviewer: This description is too long. Also, can this be useful for more things than just testing? It would be nice to use that as an example instead.

```suggestion
| `plugin` | Shows how to call a python extension module from Rust. |
```

Alternatively, you could focus on testing instead:

```suggestion
| `plugin` | Testing your python extension module from Rust. |
```

Author: I think the point was missed. The example is about making plugins in python and PyO3 for Rust application. It is not just about how  to test them (but also about that as if you just pile everything into one crate it becomes a nightmare to test). Imagine if you had to start an entire video editing software just to test a one line change in your python API. That would be a total disaster.

> | `plugin` | Illustrates how to make Python plugins work within a Rust application in a way that allows one to separate the Python API from the main application. |

Reviewer: > By plugin I mean a literal plugin you'd put in a bigger application. Specifically, if you are building something like blender video editor, you would want to have the "main" application in rust, but then also have some scripting API as well. For that to work, you would normally want to define a bunch of your application-specific interface classes, such that Python plugins can construct complex objects (in example it is the Gadget pyclass) and then pass them into the Rust code. It is much more efficient than parsing complex data structures from Python classes, and also allows you to test the interface properly before integration into the main app.

Ah, I get it 👍  I still think the description is too long though, how about

```suggestion
| `plugin` | Illustrates how to use Python as a scripting language within a Rust application |
```
